### PR TITLE
Private/caolan/can elide a dns lookup

### PR DESCRIPTION
**Pull request by caolanm**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] I have run `make prettier-write` and formatted the code.
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: net/Socket.cpp [hunk 1]
```diff
@@ -806,7 +806,7 @@ bool SocketPoll::insertNewUnixSocket(
 
     std::shared_ptr<StreamSocket> socket
         = StreamSocket::create<StreamSocket>(std::string(), fd, Socket::Type::Unix,
-                                             true, websocketHandler);
+                                             true, false, websocketHandler);
```
Comment (Ashod) [inline net/Socket.cpp:809]: I worry that multiple boolean arguments would be error prone.

Inline comments with the names would help, though verbose. Ideally, these would be enums.

**Review by Ashod — APPROVED** [review]
Thanks, @caolanm. Nice cleanup.

File: net/Socket.cpp (hunk 1 above)
Comment (caolanm) [reply net/Socket.cpp]: made this an enum now

File: net/NetUtil.cpp [hunk 2]
```diff
@@ -226,17 +226,22 @@ bool isLocalhost(const std::string& targetHost)
...
         }
     }
 
-    LOG_TRC("Host [" << targetHost << "] is not on the same host as the client: \"" << targetAddress
+    LOG_TRC("Host [" << _requestName << "] is not on the same host as the client: \"" << targetAddress
                      << "\".");
     return false;
 }
 
+bool isLocalhost(const std::string& targetHost)
```
[the remark this answers is not in the document]
Comment (caolanm) [reply net/NetUtil.cpp:240]: I'll punt this for a later time, Ideally I could remove net::isLocalhost (and net::resolveHostAddress, net::resolveAddresses) which are used by wsd/Storage.cpp with an async DNS solution.

File: net/Socket.cpp (hunk 1 above)
Comment (Ashod) [reply net/Socket.cpp]: Thanks! Looks much better.

**Pull request merged**